Views created with their respective routes

YxMARCKxY opened the pull request:
# Description
We pass the views to the format required by rails and we relate each view with its respective path.

In addition to making use of the helpers in rails such as:
    -image_tag
    -stylesheet_link_tag
    -javascript_pack_tag
    -link_to

diegoOG09 committed:
Changes switched to feature#3 branch

diegoOG09 committed:
Views created with their respective routes / Changes suggested in the CR were made

- index view added
- admin view added
- branches view added
- employees view added

- 'database.yml' file added to '.gitignore' file
- unnecessary comments deleted

Co-authored-by: armanirmr <armanirmr@gmail.com>
Co-authored-by: YxMARCKxY <mramirez73@ucol.mx>
Co-authored-by: rene3255 <ranchoeltecal@gmail.com>

YilKanda reviewed (changes requested):
Be careful, in this Pull Request commits appear from your [PR #9](https://github.com/BC-MAY-21-ROR/kata-07-reloj-checador-k7-n21-t1/pull/9/commits), you need to separate yourself from those commits, and only be above your main branch.

The commits that correspond only to the solution of the issue to be solved must appear, in this case [#10](https://github.com/BC-MAY-21-ROR/kata-07-reloj-checador-k7-n21-t1/issues/10).

Commit:
Use of 'application.html.erb' to display views / Corrected paths to 'pages' controller

- Layout controllers were deleted
- Added pages controller
- All views were moved to the folder 'pages'
- The routes to the 'pages' controller have been modified

Co-authored-by: armanirmr <armanirmr@gmail.com>
Co-authored-by: YxMARCKxY <mramirez73@ucol.mx>
Co-authored-by: rene3255 <ranchoeltecal@gmail.com>
Co-authored-by: IrenehMDev <ireneh.dev_03@icloud.com>

EduardoGHdez commented on time-clock/app/javascript/packs/timer-script.js:
```diff
@@ -0,0 +1,34 @@
+window.onload = function () {
```
It'd be nice if we can refactor this functionality to use Stimulus JS to follow the new Rails conventions

EduardoGHdez reviewed (dismissed):
LGTM

EduardoGHdez committed:
Use PG for "production"

We want to deploy our app to heroku, which does not support using SQLite
for production env.

However, we left sqlite3 for test and dev env. (for now XD)

EduardoGHdez committed:
Specify dot_image extension in admin view

EduardoGHdez committed:
Fix root/index route

EduardoGHdez committed:
Add missing extension to image references

YilKanda commented on time-clock/app/assets/stylesheets/layouts.scss:
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the layouts controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: https://sass-lang.com/
```
If you don't occupy it, delete it.

YilKanda commented on time-clock/app/helpers/index_helper.rb:
```diff
@@ -0,0 +1,2 @@
+module IndexHelper
+end
```
If you don't occupy it, delete it.

YilKanda commented on time-clock/app/helpers/layouts_helper.rb:
```diff
@@ -0,0 +1,2 @@
+module LayoutsHelper
+end
```
If you don't occupy it, delete it.

YilKanda commented on time-clock/app/views/pages/branches.html.erb:
```diff
@@ -0,0 +1,57 @@
+            <th>Address</th>
+            <th>Actions</th>
+          </thead>
+          <tr class="register">
+            <td>North</td>
+            <td>Bugambilais #132, centro Colima Col.</td>
+            <td><i class="fas fa-pencil-alt"></i> &nbsp; &nbsp;<i class="fas fa-toggle-on"></i></td>
+          </tr>
+          <tr class="register">
+            <td>North</td>
+            <td>Bugambilais #132, centro Colima Col.</td>
+            <td><i class="fas fa-pencil-alt"></i> &nbsp; &nbsp;<i class="fas fa-toggle-on"></i></td>
+          </tr>
+          <tr class="register">
+            <td>North</td>
+            <td>Bugambilais #132, centro Colima Col.</td>
+            <td><i class="fas fa-pencil-alt"></i> &nbsp; &nbsp;<i class="fas fa-toggle-on"></i></td>
+          </tr>
+          <tr class="register">
+            <td>North</td>
+            <td>Bugambilais #132, centro Colima Col.</td>
+            <td><i class="fas fa-pencil-alt"></i> &nbsp; &nbsp;<i class="fas fa-toggle-on"></i></td>
+          </tr>
+          </tr>
+          <tr class="register">
+            <td>North</td>
+            <td>Bugambilais #132, centro Colima Col.</td>
+            <td><i class="fas fa-pencil-alt"></i> &nbsp; &nbsp;<i class="fas fa-toggle-on"></i></td>
+          </tr>
```
Is this repeated code necessary?

YilKanda commented on time-clock/app/views/pages/employees.html.erb:
```diff
@@ -0,0 +1,81 @@
+            <td>312-111-11-55</td>
+            <td>Branch 1</td>
+            <td><i class="fas fa-pencil-alt"></i> &nbsp; &nbsp;<i class="fas fa-toggle-on"></i></td>
+          </tr>
+          <tr class="register">
+            <td>Marcelo</td>
+            <td>Developer</td>
+            <td>26359</td>
+            <td>marck_@hotmail.com</td>
+            <td>312-111-11-55</td>
+            <td>Branch 1</td>
+            <td><i class="fas fa-pencil-alt"></i> &nbsp; &nbsp;<i class="fas fa-toggle-on"></i></td>
+          </tr>
+          <tr class="register">
+            <td>Marcelo</td>
+            <td>Developer</td>
+            <td>26359</td>
+            <td>marck_@hotmail.com</td>
+            <td>312-111-11-55</td>
+            <td>Branch 1</td>
+            <td><i class="fas fa-pencil-alt"></i> &nbsp; &nbsp;<i class="fas fa-toggle-on"></i></td>
+          </tr>
+          <tr class="register">
+            <td>Marcelo</td>
+            <td>Developer</td>
+            <td>26359</td>
+            <td>marck_@hotmail.com</td>
+            <td>312-111-11-55</td>
+            <td>Branch 1</td>
+            <td><i class="fas fa-pencil-alt"></i> &nbsp; &nbsp;<i class="fas fa-toggle-on"></i></td>
+          </tr>
+          <tr class="register">
+            <td>Marcelo</td>
+            <td>Developer</td>
+            <td>26359</td>
+            <td>marck_@hotmail.com</td>
+            <td>312-111-11-55</td>
+            <td>Branch 1</td>
+            <td><i class="fas fa-pencil-alt"></i> &nbsp; &nbsp;<i class="fas fa-toggle-on"></i></td>
+          </tr>
```
Is this repeated code necessary?

YilKanda commented on time-clock/config/database.yml:
```diff
@@ -22,4 +22,6 @@ test:
 
 production:
   <<: *default
-  database: db/production.sqlite3
+  adapter: postgresql
+  database: database_production
+  timeout: 5000
```
It is not recommended to leave this file on GitHub, I recommend you place it in the .gitignore, and instead replace it with a database.yml.example file.

Commit:
The changes suggested in the RC were made

- Repeated code was deleted
- Unused comments were deleted
- Modified 'database.yml.example' file to add postgresql
- Changed the path of the '.gitignore' file to take into account the 'database.yml' file because before it worked but now it keeps adding it.

Co-authored-by: armanirmr <armanirmr@gmail.com>
Co-authored-by: YxMARCKxY <mramirez73@ucol.mx>
Co-authored-by: rene3255 <ranchoeltecal@gmail.com>
Co-authored-by: IrenehMDev <ireneh.dev_03@icloud.com>

EduardoGHdez commented on time-clock/Gemfile.lock:
```diff
@@ -230,6 +230,7 @@ DEPENDENCIES
   selenium-webdriver
   spring
   sqlite3 (~> 1.4)
+  stimulus-rails
```
I think you need to run `bundle` again, so the lock gets updated and this dependency gets removed 👍🏽

EduardoGHdez commented on time-clock/.gitignore:
```diff
@@ -5,7 +5,7 @@
 
 #Ignore database config
-config/database.yml
+./config/database.yml
```
Is this actually working?
I mean, I just made some changes to this file, and adding the dot won't make anything I think

EduardoGHdez reviewed (approved):
Small comments, none of them should be considered as a blocker 👍🏽 

LGTM
